app/controllers/subnets_controller.rb

```diff
@@ -40,21 +40,23 @@ def destroy
 
   # query our subnet dhcp proxy for an unused IP
   def freeip
-    not_found and return unless (s=params[:subnet_id].to_i) > 0
+    invalid_request and return unless (s=params[:subnet_id].to_i) > 0
     organization = params[:organization_id].blank? ? nil : Organization.find(params[:organization_id])
     location = params[:location_id].blank? ? nil : Location.find(params[:location_id])
     Taxonomy.as_taxonomy organization, location do
-      not_found and return unless (subnet = Subnet.authorized(:view_subnets).find(s))
+      unless (subnet = Subnet.authorized(:view_subnets).find(s))
+        render_403 #returns false, cannot put it in a oneliner
```

Comment:
Why 403? `not_found` is what Foreman is expected to return when it cannot find a resource (in this case Subnets). I think it's not possible to even fail the `.authorized` check here through the UI, as in order to call freeip you first have to select a subnet.